### PR TITLE
 added jenga API base, tests

### PR DESCRIPTION
**Pull request by manuelgeek**
## What is the Purpose?

- Added Jenga base
- Reformatted the Token Server to store multiple tokens

## Small issue
still getting 401 for the token auth, will follow up with the Jenga Team.

### CODE REVIEW

**Comment by manuelgeek** [comment]
the failing pipeline will be solved when we'll start using the janga_base more

**Review by lenileiro — APPROVED** [review]
🚀🚀🚀

File: lib/ex_pesa/Jenga/jenga_base.ex [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        end
+
+      :error ->
+        generate_token(client)
+    end
+  end
+
+  defp generate_token(client) do
+    case Tesla.post(client, "/token", %{
+           "username" => Application.get_env(:ex_pesa, :jenga)[:username],
```
Comment (lenileiro) [inline lib/ex_pesa/Jenga/jenga_base.ex:40]:
```suggestion
           username: Application.get_env(:ex_pesa, :jenga)[:username],
```

File: lib/ex_pesa/Jenga/jenga_base.ex [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+           "password" => Application.get_env(:ex_pesa, :jenga)[:password]
```
Comment (lenileiro) [inline lib/ex_pesa/Jenga/jenga_base.ex:41]:
```suggestion
           password: Application.get_env(:ex_pesa, :jenga)[:password]
```

**Review by lenileiro — CHANGES_REQUESTED** [review]

**Comment by lenileiro** [comment]
am getting a pattern matching error on the get_token function @manuelgeek

**Comment by lenileiro** [comment]
@manuelgeek I have found the error. It seems I had configured my config files incorrectly.

**Review by lenileiro — APPROVED** [review]

**Pull request merged**